iam.tf

```diff
@@ -37,3 +37,77 @@ resource "aws_iam_role_policy_attachment" "route53" {
   policy_arn = aws_iam_policy.route53[0].arn
   role       = lookup(module.eks.info.nodes.roles[count.index], "name")
 }
+
+
+locals {
+  create_eks_role_name = "${var.deploy_id}-create-eks"
+}
+
+data "aws_iam_policy_document" "create_eks_role" {
+  statement {
+    sid = "EKSDeployerEKS"
+    actions = [
+      "eks:*"
+    ]
+    resources = [
+      "arn:${data.aws_partition.current.partition}:eks:${var.region}:${local.aws_account_id}:cluster/${var.deploy_id}",
+      "arn:${data.aws_partition.current.partition}:eks:${var.region}:${local.aws_account_id}:cluster/${var.deploy_id}/*"
+    ]
+    effect = "Allow"
+  }
+
+  statement {
+    sid = "EKSDeployerKMS"
+    actions = [
+      "kms:CreateGrant",
+      "kms:DescribeKey"
+    ]
+    resources = ["arn:${data.aws_partition.current.partition}:kms:${var.region}:${local.aws_account_id}:key/*"]
+    effect    = "Allow"
+  }
+
+  statement {
+    sid = "EKSDeployerIAM"
+    actions = [
+      "iam:PassRole",
+      "iam:GetRole"
+    ]
+    resources = ["arn:${data.aws_partition.current.partition}:iam::${local.aws_account_id}:role/${var.deploy_id}-*"]
+    effect    = "Allow"
+  }
+}
+
+resource "aws_iam_policy" "create_eks_role" {
```

Comment:
this is gonna be a problem in the same account.
how do we deploy 2 deployments?

Reply:
I think the `local.create_eks_role_name` below should cover that?

Reply:
yes I missed the name was a local var.